File: Operation_Systems/lr1/lr1.py
def is_valid_hex(hex_num):
    valid_chars = set('0123456789abcdefABCDEF')
    return set(hex_num).issubset(valid_chars)

def continueProgram():
    print("Enter 'y' to continue: ", end=" ")
    if input() == 'y':
        return True
    return False


while True:
    hex_num = input("\nHexadecimal value: ")
    if hex_num.startswith('0x'):
        hex_num = hex_num[2:]

    if is_valid_hex(hex_num):
        dec_num = int(hex_num, 16)
        print("Decimal value: ", dec_num)
    else:
        print("Input error. Retry input")

    if not continueProgram():
        break

File: Operation_Systems/lr1/lr1.cpp
// Variant 3
#include <iostream>
#include <iomanip>


bool continueProgram() {
    std::cout << "Enter 'y' to continue: ";
    char choice;
    std::cin >> choice;
    if( (choice == 'y' || choice == 'Y') && std::cin.peek() == '\n')
        return true;
    else return false;
}


int main() {
    int input;

    do {
        std::cout << "\nHexadecimal value: ";
        std::cin >> std::setbase(16);
        
        while (!(std::cin >> input) || std::cin.peek() != '\n') {
            std::cin.clear();
            while (std::cin.get() != '\n');
            std::cout << "Input error. Retry input\n";    
        }
        std::cout << "Decimal value: " << std::setbase(10) << input << std::endl;

    } while (continueProgram());

    return 0;
}
